sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -54,6 +54,11 @@ object JDBCRDD extends Logging {
     val table = options.tableOrQuery
     val dialect = JdbcDialects.get(url)
     val conn: Connection = JdbcUtils.createConnectionFactory(options)()
+
+    if (null == conn) {
```

Comment:
Nit: `conn == null`. Yes, I think this check should be in the function that `createConnectionFactory` returns. I doubt it's ever reasonable to return a null connection.

Reply:
Yea. It should be in that function. We can use `require` function to throw illegal argument exception. Looks it can potentially return `null` assuming from it's doc.

Reply:
Does it mean that I should to replace this check to `createConnectionFactory` ?

Reply:
Yeah, to the lambda it returns